redis/src/main/scala/zio/redis/options/Streams.scala

```diff
@@ -54,4 +73,72 @@ trait Streams {
   type NoAck = NoAck.type
 
   case class MaxLen(approximate: Boolean, count: Long)
+
+  case class StreamEntry(id: String, fields: Map[String, String])
+
+  case class StreamInfo(
+    length: Option[Long],
+    radixTreeKeys: Option[Long],
+    radixTreeNodes: Option[Long],
+    groups: Option[Long],
+    lastGeneratedId: Option[String],
+    firstEntry: Option[StreamEntry],
+    lastEntry: Option[StreamEntry],
+    undefined: Map[String, RespValue] = Map.empty
+  )
+
+  object StreamInfo {
+    lazy val empty: StreamInfo = StreamInfo(None, None, None, None, None, None, None)
+  }
+
+  case class StreamGroupInfo(
+    name: Option[String],
+    consumers: Option[Long],
+    pending: Option[Long],
+    lastDeliveredId: Option[String],
+    undefined: Map[String, RespValue] = Map.empty
+  )
+
+  object StreamGroupInfo {
+    lazy val empty: StreamGroupInfo = StreamGroupInfo(None, None, None, None)
+  }
+
+  case class StreamConsumerInfo(
+    name: Option[String],
+    idle: Option[Long],
+    pending: Option[Long],
+    undefined: Map[String, RespValue] = Map.empty
+  )
+
+  object StreamConsumerInfo {
+    lazy val empty: StreamConsumerInfo = StreamConsumerInfo(None, None, None)
+  }
+
+  object XInfoFields {
+    // The return structures of the same key in different commands are completely incompatible.
+    // So key can be shared, but the output cannot be shared.
+    // https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/StreamConsumersInfo.java
+    val NAME: String    = "name"
+    val IDLE: String    = "idle"
+    val PENDING: String = "pending"
+
+    // https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/StreamGroupInfo.java
+    val CONSUMERS: String      = "consumers"
+    val LAST_DELIVERED: String = "last-delivered-id"
+
+    // https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/StreamInfo.java
+    val LENGTH: String            = "length"
+    val RADIX_TREE_KEYS: String   = "radix-tree-keys"
+    val RADIX_TREE_NODES: String  = "radix-tree-nodes"
+    val GROUPS: String            = "groups"
+    val LAST_GENERATED_ID: String = "last-generated-id"
+    val FIRST_ENTRY: String       = "first-entry"
+    val LAST_ENTRY: String        = "last-entry"
+
+    // TODO STREAMS FULL
+    val ENTRIES: String   = "entries"
+    val PEL_COUNT: String = "pel-count"
+    val SEEN_TIME: String = "seen-time"
+  }
```

Comment:
Write constants in `PascalCase` and you can remove these comments. Also, I suggest that you make this object private on `redis` package level because the end-user doesn't need to see these.